File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = {};
        
        ArrayList<Integer> al = new ArrayList<>();
        
        for(int[] com: commands) {
            int[] tmpArr = new int[com[1]-com[0]+1];
            System.arraycopy(array, com[0]-1, tmpArr, 0, com[1]-com[0]+1);
            // tmpArr = Arrays.copyOfRange(array, com[0]-1, com[1]);
            Arrays.sort(tmpArr);
            al.add(tmpArr[com[2]-1]);
        }
        
        answer = al.stream().mapToInt(i->i).toArray();
        
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.js
function solution(array, commands) {
    var answer = [];
    
    commands.forEach((e) => {
        // console.log("e["+e[0]+", "+e[1]+", "+e[2]+"]")
        const arr = array.slice(e[0]-1, e[1]).sort((a, b) => a - b)
        // console.log(arr)
        answer.push(arr[e[2]-1])
    })
    return answer;
}